treeherder/webapp/api/jobs.py

```diff
@@ -96,9 +96,73 @@ class Meta:
         }
 
 
-class JobsViewSet(viewsets.ViewSet):
+class JobsViewSet(viewsets.ReadOnlyModelViewSet):
     """
-    This viewset is responsible for the jobs endpoint.
+    This viewset is the jobs endpoint.
+    """
+    _default_select_related = [
+        'job_type',
+        'job_group',
+        'machine_platform',
+        'signature',
+    ]
+    _query_field_names = [
+        'submit_time',
+        'start_time',
+        'end_time',
+        'failure_classification_id',
+        'id',
+        'job_group__name',
+        'job_group__symbol',
+        'job_type__name',
+        'job_type__symbol',
+        'last_modified',
+        'option_collection_hash',
+        'machine_platform__platform',
+        'option_collection_hash',
+        'push_id',
+        'result',
+        'signature__signature',
+        'state',
+        'tier',
+    ]
+    _output_field_names = [
+        'failure_classification_id',
+        'id',
+        'job_group_name',
+        'job_group_symbol',
+        'job_type_name',
+        'job_type_symbol',
+        'last_modified',
+        'platform',
+        'push_id',
+        'result',
+        'signature',
+        'state',
+        'tier',
+        'duration',
+        'platform_option',
+    ]
+    queryset = Job.objects.all().order_by('id').select_related(
+        *_default_select_related
+    ).values(*_query_field_names)
+    serializer_class = serializers.JobSerializer
+    filterset_class = JobFilter
+    pagination_class = pagination.JobPagination
+
+    def get_serializer_context(self):
+        option_collection_map = OptionCollection.objects.get_option_collection_map()
```

Comment:
Is this getting called for every iteration over the queryset, or is it called once and then cached?

Reply:
It's called once every time the ``/api/job/`` endpoint is called.  The result list of options *rarely* changes, but it does from time to time.  So we could possibly cache this in ``redis``.  Perhaps invalidate it when a ``create`` is done on the table.  I'll see if that's straightforward...